Contents 
1. Import libraries
2. Check statistics 
3. Check for mixed type data
4. Check for missing values
5. Check for duplicates 
6. Export data 

1. Import libraries

In [1]:
#import libraries
import pandas as pd 
import numpy as np
import os 

In [2]:
path = '/Users/wenboyang/Desktop/Data Analytics/05-08-2023 Instacart Basket Analysis /02 Data/'

In [3]:
orders_data = os.path.join(path, 'Prepared Data', 'orders_wrangled.csv')

In [4]:
df_ords = pd.read_csv(orders_data, index_col=False)

2. Check statistics 

In [5]:
df_ords.describe()

Unnamed: 0      order_id       user_id  order_number  \
count  3.421083e+06  3.421083e+06  3.421083e+06  3.421083e+06   
mean   1.710541e+06  1.710542e+06  1.029782e+05  1.715486e+01   
std    9.875817e+05  9.875817e+05  5.953372e+04  1.773316e+01   
min    0.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00   
25%    8.552705e+05  8.552715e+05  5.139400e+04  5.000000e+00   
50%    1.710541e+06  1.710542e+06  1.026890e+05  1.100000e+01   
75%    2.565812e+06  2.565812e+06  1.543850e+05  2.300000e+01   
max    3.421082e+06  3.421083e+06  2.062090e+05  1.000000e+02   

       orders_day_of_week  order_hour_of_day  days_since_prior_order  
count        3.421083e+06       3.421083e+06            3.214874e+06  
mean         2.776219e+00       1.345202e+01            1.111484e+01  
std          2.046829e+00       4.226088e+00            9.206737e+00  
min          0.000000e+00       0.000000e+00            0.000000e+00  
25%          1.000000e+00       1.000000e+01            4.000000e+00  
50%          3.000000e+00       1.300000e+01            7.000000e+00  
75%          5.000000e+00       1.600000e+01            1.500000e+01  
max          6.000000e+00       2.300000e+01            3.000000e+01

3. Check for mixed data

In [6]:
#Check for mixed columns 
for col in df_ords.columns.tolist():
  weird = (df_ords[[col]].applymap(type) != df_ords[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df_ords[weird]) > 0:
    print (col)

No mixed columns found

4. Check missing values 

In [7]:
#5 Check for missing values 
df_ords.isnull().sum()

Unnamed: 0                     0
order_id                       0
user_id                        0
order_number                   0
orders_day_of_week             0
order_hour_of_day              0
days_since_prior_order    206209
dtype: int64

There are 206209 missing values in the days_since_prior_order column. Since there are so many missing values I would not remove them. Instead I would leave 
them as is or flag them with N/A. 


5. Check for duplicates

In [8]:
#Find full duplicates
df_dups = df_ords[df_ords.duplicated()]

In [9]:
df_dups

Empty DataFrame
Columns: [Unnamed: 0, order_id, user_id, order_number, orders_day_of_week, order_hour_of_day, days_since_prior_order]
Index: []

No duplicates found

6. Export data

In [11]:
#Export df_ords

df_ords.to_csv(os.path.join(path,'Prepared Data', 'ords_checked.csv'))